File: jscomp/test/fs_test.ml
let suites :  Mt.pair_suites ref  = ref []
let test_id = ref 0
let eq loc (x, y) =
  incr test_id ;
  suites :=
    (loc ^" id " ^ (string_of_int !test_id), (fun _ -> Mt.Eq(x,y))) :: !suites


external readFileSync : string -> ([`utf8 | `ascii] [@bs.string]) -> string
  = "" [@@bs.module "fs"]


type watcher

external watch : unit -> watcher = "" [@@bs.module "fs"]

type error

external on :
  watcher ->

  (
    [
      | `change of (string (* event*) -> string (*filename*) -> unit )
      | `error of (error -> unit [@bs]) ]
      [@bs.string]
  ) -> unit = "on" [@@bs.send]

open Node
let () =
  let current_file : string =
    match [%node __filename] with
    | Some x -> x
    | None -> "<Not Node JS>"  in
  let current_dir_name : string =
    match  [%node __dirname] with
    | Some x -> x
    | None -> "<Not Node Js>"  in
  let _content = readFileSync current_file `utf8 in
  let _file_list = Fs.readdirSync current_dir_name in
  let pathobj =   Path.parse current_dir_name in
  match  [%node _module] with
  | Some module_ ->
    Js.log (module_##id, module_##paths) ;
    eq __LOC__ (pathobj##name, "test" )
  | None -> ()



let () =
  (* Js.log ("ARGV", Node.Process.process##argv);   *)
  Mt.from_pair_suites __MODULE__ !suites

File: jscomp/test/fs_test.js
'use strict';

var Mt = require("./mt.js");
var Fs = require("fs");
var Path = require("path");
var Block = require("../../lib/js/block.js");

var suites = {
  contents: /* [] */0
};

var test_id = {
  contents: 0
};

function eq(loc, param) {
  var y = param[1];
  var x = param[0];
  test_id.contents = test_id.contents + 1 | 0;
  suites.contents = /* :: */{
    _0: /* tuple */[
      loc + (" id " + String(test_id.contents)),
      (function (param) {
          return {
                  tag: /* Eq */0,
                  _0: x,
                  _1: y
                };
        })
    ],
    _1: suites.contents
  };
  
}

var x = typeof __filename === "undefined" ? undefined : __filename;

var current_file = x !== undefined ? x : "<Not Node JS>";

var x$1 = typeof __dirname === "undefined" ? undefined : __dirname;

var current_dir_name = x$1 !== undefined ? x$1 : "<Not Node Js>";

Fs.readFileSync(current_file, "utf8");

Fs.readdirSync(current_dir_name);

var pathobj = Path.parse(current_dir_name);

var module_ = typeof module === "undefined" ? undefined : module;

if (module_ !== undefined) {
  console.log(/* tuple */[
        module_.id,
        module_.paths
      ]);
  eq("File \"fs_test.ml\", line 45, characters 7-14", /* tuple */[
        pathobj.name,
        "test"
      ]);
}

Mt.from_pair_suites("Fs_test", suites.contents);

exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
/* x Not a pure module */
